src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -441,7 +441,10 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
                     var transactionItems = new List<TransactionItemModel>();
 
                     // Sorting the history items by descending dates. That includes received and sent dates.
-                    List<FlatHistory> items = accountHistory.History.OrderByDescending(o => o.Transaction.SpendingDetails?.CreationTime ?? o.Transaction.CreationTime).ToList();
+                    List<FlatHistory> items = accountHistory.History
+                                                            .OrderBy(o => (o.Transaction.SpendingDetails?.IsSpentConfirmed() ?? true) ? 1 : 0)
```

Comment:
I still don't think this is quite correct - consider the various cases (there are a lot more than first meets the eye):

Transaction spent (i.e. `SpendingDetails` is not `null`)
a. Transaction itself is confirmed (i.e. `Transaction.BlockHash` is not `null`)
	a1. Spending details are confirmed (i.e. the block data is populated)
	a2. Spending details are unconfirmed
b. Transaction itself is unconfirmed
    b1. Spending details are confirmed (this case should not be possible, enforced via consensus)
	b2. Spending details unconfirmed

Transaction not spent (i.e. `SpendingDetails` is `null`)
c. Transaction itself is confirmed
d. Transaction itself is unconfirmed


Then the values assigned to each category within the LINQ:

a1. 1
a2. 0 (this is wrong, the underlying transaction is confirmed and should be in the other group)
b1. 1
b2. 0 (this is correct but by accident)
c. 1 (due to null coalesce, but not because the transaction itself is confirmed - this seems wrong too)
d. 1 (this is wrong)

Unconfirmed transactions should get 0 so that we get a list that looks like the following:

['0' group]
Newer unconfirmed tx
Older unconfirmed tx

['1' group]
Newer confirmed tx
Older confirmed tx

Note the .IsSpent() and .IsConfirmed() methods on TransactionData itself, which may be more fruitful to use (IsSpentConfirmed is only in the context of the spending tx, not the original tx).

Also, perhaps put some of the above in as code comments so that the scenarios are documented :D

Reply:
@zeptin Following all the scenarios listed above and considering that (b1) is not possible by consensus, I think it can all be simplified to `.OrderBy(o => o.Transaction.IsConfirmed() ? 1 : 0)`, am I missing something?

Reply:
Yeah, I think that should be sufficient